logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineInput.java

```diff
@@ -28,13 +28,65 @@
  * Represents the in endpoint of a pipeline to pipeline communication.
  * */
 public interface PipelineInput {
+
+    enum ReceiveStatus {CLOSING, COMPLETED, FAIL}
+
+    final class ReceiveResponse {
+        private final ReceiveStatus status;
+        private final Integer sequencePosition;
+        private final Throwable cause;
+
+        static ReceiveResponse closing() {
+            return new ReceiveResponse(ReceiveStatus.CLOSING);
```

Comment:
since the status is immutable anything against making it return a static`return ReceiveResponse.CLOSING` ?
(same for `completed()` method)

Reply:
For statuses `CLOSING` and `COMPLETED` I would agree but for `FAIL`, and this is the reason for this class to exists, is carry also the context (exception and position) when the error was thrown.